packages/veritone-widgets/src/redux/modules/multipleEngineSelection/index.js

```diff
@@ -0,0 +1,223 @@
+      nextState[id].selected = selected;
+      return nextState;
+    } else {
+      // toggle all pickers
+      let nextState = { ...state };
+      const ids = Object.keys(nextState);
+      ids.forEach(id => {
+        nextState[id].selected = toggleSelection(nextState, id, rowId);
+      });
+      return nextState;
+    }
+  },
+  [FETCH_ENGINES_SUCCESS](state, action) {
+    let engines = get(state, [getPickerID(action), 'engines']) || [];
+    let enginesNextPage = get(action, 'payload.engines.records');
+
+    let myOrgId = get(action, 'payload.me.organizationId');
+    enginesNextPage.forEach(
+      engine => (engine.owned = engine.ownerOrganization.id === myOrgId)
+    );
+
+    engines = engines.concat(enginesNextPage);
+    let lastSearch = get(state, [getPickerID(action), 'lastSearch']);
+    if (
+      isNil(lastSearch) ||
+      lastSearch === get(action, 'meta.variables.search')
+    ) {
+      return {
+        ...state,
+        [getPickerID(action)]: {
+          loadingEngines: false,
+          loadingEnginesFailed: false,
+          engines,
+          offset:
+            get(action, 'meta.variables.offset') +
+            get(action, 'payload.engines.count'),
+          lastCount: get(action, 'payload.engines.count'),
+          lastSearch,
+          selected: get(state, [getPickerID(action), 'selected'])
+        }
+      };
+    } else {
+      return state;
+    }
+  },
+  [FETCH_ENGINES_FAILURE](state, action) {
+    return {
+      ...state,
+      [getPickerID(action)]: {
+        loadingEngines: false,
+        loadingEnginesFailed: true,
+        engines: [],
+        offset: 0,
+        lastCount: 0,
+        lastSearch: undefined,
+        selected: {}
+      }
+    };
+  }
+});
+
+export default reducer;
+
+function local(state) {
+  return state[namespace];
+}
+
+export const toggleEngine = ({ id, rowId }) => {
+  return {
+    type: TOGGLE_ENGINE,
+    id,
+    rowId
+  };
+};
+
+export const fetchEngines = ({ category, id, owned, engineName }) => async (
+  dispatch,
+  getState
+) => {
+  const lastSearch = get(local(getState()), [id, 'lastSearch']);
+  let offset = get(local(getState()), [id, 'offset']) || 0;
+  if (!isEqual(engineName, lastSearch)) {
+    offset = 0;
+  }
+
+  const query = `
+    query {
+      engines(offset: ${offset}, limit: ${PAGE_SIZE}, category: "${category}", name: "${engineName}", state: active, orderBy: {field: name, direction: asc}, owned: ${owned ===
+    true}) {
+        records {
+          id,
+          name,
+          displayName,
+          logoPath,
+          builds {
+            records {
+              id,
+              name,
+              version,
+              price
+            }
+          },
+          description,
+          deployedVersion,
+          ownerOrganization {
+            name,
+            id
+          }
+        },
+        count
+      },
+      me {
+        organizationId
+      }
+    }
+  `;
+
+  return await callGraphQLApi({
```

Comment:
not a real issue but you can omit `await` if you're returning from an asynchronous operation inside an async function